### ISSUE TITLE
Set license for API requests

### ISSUE THREAD

**Issue by marco-gallardo** [alexreisner/geocoder#1620]
### Context:
_Smarty_ offers different licenses that determine a set of available services, usage details, etc. Not being able to specify this _license_ limits our integration with _Smarty_ and prevents taking full advantage of _Smarty_'s features.

### Expected behavior
It'd be great to be available to provide _license_ argument that ends up being passed to _Smarty_ API:
```
Geocoder.configure(..., license: '<license>')
```
I reviewed geocoder's code and apparently this will require only a few files to be changed:
- include _license_ on [options](https://github.com/alexreisner/geocoder/blob/cb9a1cb16418b507e4edaa6f711e78acb4f03e62/lib/geocoder/configuration.rb#LL53C2-L53C2)
- include _license_ on [set_defaults](https://github.com/alexreisner/geocoder/blob/cb9a1cb16418b507e4edaa6f711e78acb4f03e62/lib/geocoder/configuration.rb#L102) method
- register _license_ in [query_url_params](https://github.com/alexreisner/geocoder/blob/cb9a1cb16418b507e4edaa6f711e78acb4f03e62/lib/geocoder/lookups/smarty_streets.rb#L39) method

Any thoughts about this?

**Comment by alexreisner** [comment]
Yes, thanks, this sounds great, and it's actually even easier than you think. You can pass arbitrary parameters to specific lookups like this:

    Geocoder.configure(..., smarty_streets: {license: '<license>'})

So there's no need to modify `options` or `set_defaults`. Just add the license parameter in `query_url_params`, where you can access it via `configuration[:license]` (`configuration` returns all config for the current lookup, merged into a single hash). You might also want to mention this in a "Notes" item in [the Smarty documentation](https://github.com/alexreisner/geocoder/blob/master/README_API_GUIDE.md#smartystreets-smarty_streets).
